demo/movielens-1m-ps/movielens-1m-ps.py

```diff
@@ -0,0 +1,214 @@
+  worker_list = config['cluster']['worker']
+
+  num_ps_tasks = len(ps_list)
+  num_worker_tasks = len(worker_list)
+  sess_config = tf.compat.v1.ConfigProto()
+  sess_config.intra_op_parallelism_threads = 1
+  sess_config.inter_op_parallelism_threads = 1
+  cluster = tf.train.ClusterSpec(config['cluster'])
+  server = tf.distribute.Server(cluster,
+                                protocol="grpc",
+                                job_name="worker",
+                                task_index=worker_id,
+                                config=sess_config)
+  with tf.compat.v1.device("/job:worker/replica:0/task:{}".format(worker_id)):
+    trainer = Trainer(worker_id=worker_id,
+                      worker_num=num_worker_tasks,
+                      ps_num=num_ps_tasks,
+                      batch_size=64,
+                      ckpt_dir=None)
+    train_iter = trainer.read_batch()
+    train_data = train_iter.get_next()
+
+  device_setter = tf.compat.v1.train.replica_device_setter(
+      ps_tasks=num_ps_tasks,
+      worker_device="/job:worker/replica:0/task:{}".format(worker_id),
+      ps_device="/job:ps")
+
+  with tf.compat.v1.device(device_setter):
+    outputs = trainer.build_graph(train_data)
+
+  with tf.compat.v1.train.MonitoredTrainingSession(
+      master=server.target,
+      is_chief=FLAGS.is_chief,
+      checkpoint_dir=trainer.ckpt_dir if FLAGS.is_chief else None,
+      config=sess_config,
+  ) as sess:
+    sess.run([train_iter.initializer])
+
+    step = 0
+    while True:
+      step += 1
+      try:
+        _, _loss, _pred = sess.run(
+            [outputs["update"], outputs["loss"], outputs["predict"]])
+
+        _size = sess.run(outputs["size"])
+        if step % 100 == 0:
+          print("[worker{}]step{}:\tloss={:.4f}\t size={}".format(
+              worker_id, step, float(_loss), _size))
+      except tf.errors.OutOfRangeError:
+        print("[worker{}]no more data!".format(worker_id))
+        break
+
+
+def start_ps(task_id, config):
+  print("ps config", config)
+  cluster = tf.train.ClusterSpec(config["cluster"])
+
+  sess_config = tf.compat.v1.ConfigProto()
+  sess_config.intra_op_parallelism_threads = 1
+  sess_config.inter_op_parallelism_threads = 1
+  server = tf.distribute.Server(cluster,
+                                config=sess_config,
+                                protocol='grpc',
+                                job_name="ps",
+                                task_index=task_id)
+  server.join()
+
+
+def main(argv):
+  ps_list = FLAGS.ps_list.replace(' ', '').split(',')
+  worker_list = FLAGS.worker_list.replace(' ', '').split(',')
+  task_mode = FLAGS.task_mode
+  task_id = FLAGS.task_id
+
+  print('ps_list: ', ps_list)
+  print('worker_list: ', worker_list)
+
+  if task_mode == 'ps':
+    ps_config = {"cluster": {"ps": ps_list, "worker": worker_list}}
+    start_ps(task_id, ps_config)
+  elif task_mode == 'worker':
+    worker_config = {"cluster": {"ps": ps_list, "worker": worker_list}}
```

Comment:
`ps_config` and `worker_config` seem to be the same. Use only `cluster_config` instead.

Reply:
accept